resources/PythonPip3Dsc/PythonPip3Dsc.psm1

```diff
@@ -4,6 +4,77 @@
 using namespace System.Collections.Generic
 
 #region Functions
+function Invoke-Process
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$FilePath,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]$ArgumentList,
+
+        [ValidateSet("Full", "StdOut", "StdErr", "ExitCode", "None")]
+        [string]$DisplayLevel
+    )
+
+    try
+    {
+        $pinfo = New-Object System.Diagnostics.ProcessStartInfo
+        $pinfo.FileName = $FilePath
+        $pinfo.RedirectStandardError = $true
+        $pinfo.RedirectStandardOutput = $true
+        $pinfo.UseShellExecute = $false
+        $pinfo.WindowStyle = 'Hidden'
+        $pinfo.CreateNoWindow = $true
+        $pinfo.Arguments = $ArgumentList
+        $p = New-Object System.Diagnostics.Process
+        $p.StartInfo = $pinfo
+        $p.Start() | Out-Null
+
+        $stOut = @()
+        while (-not $p.StandardOutput.EndOfStream)
+        {
+            $stOut += $p.StandardOutput.ReadLine()
+        }
+
+        $stErr = @()
+        while (-not $p.StandardError.EndOfStream)
+        {
+            $stErr += $p.StandardError.ReadLine()  
+        }
+
+        $result = [pscustomobject]@{
+            Title     = ($MyInvocation.MyCommand).Name
+            Command   = $FilePath
+            Arguments = $ArgumentList
+            StdOut    = $stOut
+            StdErr    = $stErr
+            ExitCode  = $p.ExitCode
+        }
+
+        $p.WaitForExit()
+
+        if (-not([string]::IsNullOrEmpty($DisplayLevel)))
+        {
+            switch ($DisplayLevel)
+            {
+                "Full" { return $result; break }
+                "StdOut" { return $result.StdOut; break }
+                "StdErr" { return $result.StdErr; break }
+                "ExitCode" { return $result.ExitCode; break }
+            }
+        }
```

Comment:
I don't think this is needed. Can you return the custom PS object that you created above instead of having a parameter determine the output. What if the caller wants multiple outputs like the exit code and stdout?

Reply:
Fixed!